src/internal/types.ts

```diff
@@ -50,7 +50,7 @@ interface NetInfoDisconnectedState<T extends NetInfoStateType> {
 
 export interface NetInfoUnknownState {
   type: NetInfoStateType.unknown;
-  isConnected: null;
+  isConnected: boolean;
```

Comment:
This looks a little riskier. This is the general unknown state object for all platforms, not just web.
Are you certain all other platforms will always have a valid isConnected boolean?
Your test plan mentioned testing on web but has no mention of testing on android, ios, macos, or windows - and it's important with a super-multi-platform project not to break those.

In fact, it directly conflicts with this PR
https://github.com/react-native-netinfo/react-native-netinfo/pull/444

which was to address the ambiguity in this module's initial state return:
https://github.com/react-native-netinfo/react-native-netinfo/issues/295

Perhaps a better fix would be to figure out *why* the network type was unknown vs whatever it really is, and add a new NetworkType object that actually identifies the network type and contains the real network information ?

Otherwise it's a breaking change as you've changed the type surface area, and also because it conflicts with code specifically checking for the value you made impossible :thinking: 

what do you think? Is it possible to identify those network types you are having difficulty with and type them for real?

Reply:
> Is it possible to identify those network types you are having difficulty with and type them for real?



As far as i can tell, it's impossible to identify  those network types
When connected via instant tethering, my chromebook just has type: 'unknown' in `NetworkInformation `object
![Screenshot 2021-11-28 23 28 23](https://user-images.githubusercontent.com/9059945/143786950-bff5924e-867c-418a-97ac-a2b5e065fe11.png)

> This looks a little riskier. This is the general unknown state object for all platforms, not just web.
Are you certain all other platforms will always have a valid isConnected boolean?

True, i haven't thought about that. 
Since I cannot create a new NetworkType object, I would propose to change `isConnected: boolean;` to `isConnected: boolean | null;`. similar to NetInfoConnectedState, NetInfoWifiState and NetInfoEthernetState.

Reply:
Hmm - I see what you mean: https://caniuse.com/netinfo

I believe that would make the type backwards-compatible as it's an additive change vs previous type
It should not affect the native platforms.

Two big pluses :-)

Are there *any* web cases where the null could be vaild, in your specific setup? That is, the previous work here was aimed at making "null" contain information, that it was the first hook fetch so don't rely on it.
Is it possible that on the initial fetch via hook here that it actually should be unknown/null, and then only after initial fetch it should be unknown/true|false ?

Reply:
>  Are there any web cases where the null could be vaild, in your specific setup?

In[ our specific setup](https://github.com/Expensify/App) we use isConnected as a bool, I haven't found any pieces of code that would rely on it being null.

Reply:
Thanks for checking! I'll scan it again, but if it's null|boolean, this should be okay...

Reply:
Final scan for me is always the related documentation. Can you add a chunk that takes this:

> If there is an active network connection. If unknown defaults to `null`.

and makes it this?

> Defaults to `null` on most platforms for `unknown` networks. Note: Web browsers report network type `unknown` for many otherwise valid networks (https://caniuse.com/netinfo), so i`sConnected` may be `true` or `false` and represent a real connection status even for `unknown` network types in certain cases.

Normally I'd help move things along by posting a commit on your fork but your editing on master and I won't push to someone else's main branch :fearful:  :smile: 

Thanks for collaborating on this, I think we have a good fix here for you in a moment, without breaking anyone

Reply:
Pushed the commit with changes to README.
Thanks for helping with the fix